Configuration/PyReleaseValidation/test/test_mc_setup/test_MC_setup.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+# This script mimics a MC chain with HLT run 
+# a in a different release from the current one
+
+echo '#### MC Setup'
+
+# this would make it run only in 14_0_X releases, need to keep?
+
+# if [[ ! ("$CMSSW_VERSION" == *"14_0_"*) ]]; then
+#     echo "This is not a 14_0_X release. No need to test this. Exiting."
+#     exit 0
+# fi
+
+# inputs
+conditions=$1 
+era=$2
+hlt=$3
+release_hlt=$4
+gt_hlt=$5
+beamspot=$6
+
+base_dir=$PWD
+base_cms=$CMSSW_BASE
+base_arch=$SCRAM_ARCH
+
+
+# to take into account changes in gcc
+base_arch_no_gcc=$(echo $SCRAM_ARCH | cut -d "_" -f -2)
+if [[ ! ("$CMSSW_VERSION" == $release_hlt) ]]; then
+    hlt_arch=$(echo $(ls -dtrh /cvmfs/cms.cern.ch/${base_arch_no_gcc}*/cms/cmssw/${release_hlt} | tail -1) | cut -d "/" -f 4)
+else
+    hlt_arch=$base_arch
+fi
+
+
+echo '> GT           : ' $conditions
+echo '> Era          : ' $era
+echo '> BS           : ' $beamspot
+echo '> HLT          : ' $hlt
+echo '> HLT release  : ' $release_hlt
+echo '> HLT GT       : ' $gt_hlt
+echo ' - with SCRAM_ARCH='$hlt_arch
+echo ''
+
+
+
+############################################################################################################
+# setting up the temporary working dir, to be removed once done (or crashed)
+tmp_dir=$(mktemp -d -t test_MC_${3}_${4}_${5}_XXXXXXX)
+mkdir -p $tmp_dir
+
+echo '> Working dir: ' ${tmp_dir}
+echo ''
+
+# the tmp dir cleaning 
+function job_clean()
+{
+    echo '>>>>> Cleaning up and going back to base setup <<<<<'
+    rm -rf "$tmp_dir"
+    
+    export SCRAM_ARCH=$base_arch
+    cd $CMSSW_BASE
+    eval `scram runtime -sh`
+
+    cd $base_dir
+    # export TEST_CLEANED_test_mc_23_workdir=0
+}
+
+trap job_clean EXIT
+trap job_clean SIGINT 
+
+############################################################################################################
+# GEN SIM
+# Not really needed, but anyway
+cd $CMSSW_BASE
+eval `scram runtime -sh`
+cd $tmp_dir
+
+echo $PWD
+
+bash test-MC-setup_gen_sim.sh $CMSSW_VERSION $conditions $era $beamspot # Once we have the proper GT this will be DBrealistic 
+
+############################################################################################################
+# HLT
+export SCRAM_ARCH=$hlt_arch
+cd /cvmfs/cms.cern.ch/$hlt_arch/cms/cmssw/$release_hlt/
+eval `scram runtime -sh`
+cd $tmp_dir
+
+bash test-MC-setup_hlt.sh $release_hlt $gt_hlt $era $hlt 
+export SCRAM_ARCH=$base_arch
+
+############################################################################################################
+# RECO + PAT
+
+cd $CMSSW_BASE
```

Comment:
Doesn't `$CMSSW_BASE` point to `/cvmfs/cms.cern.ch/$hlt_arch/cms/cmssw/$release_hlt/` at this point?

Reply:
Yes, fixing it.